modules/standard/Random.chpl

```diff
@@ -285,28 +285,28 @@ module Random {
 
     /*
       Advances/rewinds the stream to the `n`-th value in the sequence.
-      The first value is with n=1.
+      The first value is with n=1.  n must be > 0, otherwise an
+      IllegalArgumentError is thrown.
 
       :arg n: The position in the stream to skip to.  Must be > 0.
       :type n: `integral`
      */
-
-    proc skipToNth(n: integral) {
+    proc skipToNth(n: integral) throws {
       compilerError("RandomStreamInterface.skipToNth called");
     }
 
     /*
       Advance/rewind the stream to the `n`-th value and return it
-      (advancing the stream by one).  This is equivalent to
+      (advancing the stream by one).  n must be > 0, otherwise an
+      IllegalArgumentError is thrown.  This is equivalent to
       :proc:`skipToNth()` followed by :proc:`getNext()`.
 
       :arg n: The position in the stream to skip to.  Must be > 0.
       :type n: `integral`
 
       :returns: The `n`-th value in the random stream as type :type:`eltType`.
-    */
-
-    proc getNth(n: integral): eltType {
+     */
+    proc getNth(n: integral): eltType throws {
```

Comment:
This was intriguing to me:  Is it the case that if one overload of a function throws, they all must?

Reply:
That's true, but RandomStreamInterface exists just for documentation purposes (i.e. the other streams don't inherit from it).

Reply:
Note that this class isn't actually used for anything (so far as I know). The documentation says:

"""
The :class:`RandomStreamInterface` is included here only for documentation
and does not help with compilation in any way.  In the future, we hope to
turn it into an interface.
"""

So this class is intended to be an abstract base class/interface. In this case I think it's an override, not an overload and so far as I know `throws` is part of the signature so overridden methods must also be declared throws if the base method is.